### PR TITLE
Support for V4L2_PIX_FMT_Y16 and setting CV_CAP_PROP_FOURCC for v4l2

### LINKED ISSUE #7194: Add support for Y16 format in the V4L module
Some cameras only support the Y16 image format. However, the V4L videoio module does not support V4L2_PIX_FMT_Y16.

### PR DESCRIPTION
**Pull request by Algomorph**
Resolves #7194

1) Changed cap_v4l.cpp so the backend tries to use V4L2_PIX_FMT_Y16 (adapted from @stbnps in #7196).
2) Unlike #7196, Cameras started with Y16 (V4L2_PIX_FMT_Y16) format via v4l2 backend will now exhibit default camera behavior, i.e. convert the 16-bit image to BGR as with all other formats. 16-bit 1-channel output will now only be produced for Y16 if CV_CAP_PROP_CONVERT_RGB is explicitly set to "false" using VideoCap::set method.
3) v4l2 videoio backend now supports setting CV_CAP_PROP_FOURCC explicitly (icvSetPropertyCAM_V4L function in cap_v4l.cpp), allowing users to manually set the codec on cameras that support multiple codecs.

### CODE REVIEW

File: modules/videoio/src/cap_v4l.cpp [hunk 1]
```diff
@@ -1764,6 +1797,20 @@ static int icvSetPropertyCAM_V4L( CvCaptureCAM_V4L* capture,
         capture->convert_rgb = bool(value) && possible;
         retval = possible || !bool(value);
         break;
+    case CV_CAP_PROP_FOURCC:
+        {
+	        __u32 old_palette = capture->palette;
```
Comment (alalek) [inline modules/videoio/src/cap_v4l.cpp]: Please remove tab here

File: modules/videoio/src/cap_v4l.cpp [hunk 2]
```diff
@@ -585,11 +592,16 @@ static void v4l2_create_frame(CvCaptureCAM_V4L *capture) {
         case V4L2_PIX_FMT_YVU420:
             size.height = size.height * 3 / 2; // "1.5" channels
             break;
+        case V4L2_PIX_FMT_Y16:
+            if(!capture->convert_rgb){
+                depth = IPL_DEPTH_16U;
```
Comment (alalek) [inline modules/videoio/src/cap_v4l.cpp]: channels = 1

Comment (alalek) [inline modules/videoio/src/cap_v4l.cpp]: OK, it is above: `channels = v4l2_num_channels(...)`

File: modules/videoio/src/cap_v4l.cpp (hunk 1 above)
Comment (Algomorph) [inline modules/videoio/src/cap_v4l.cpp]: Sorry about that, IDE feature for converting indentation tabs to spaces misfired here somehow. Fixed now.

**Comment by Algomorph** [comment]
@alalek , should be good now.

**Comment by alalek** [comment]
Looks good to me!
:+1:

**Comment by Guille129** [comment]
Hi, I changed file cap_v4l.cpp as you said, and set the property CV_CAP_PROP_CONVERT_RGB to False but it is still not working. I read 8 bits depth pixels. Thank you!

	#include "opencv2/imgproc/imgproc.hpp"
	#include "opencv2/highgui/highgui.hpp"

	#include <iostream>
	#include <conio.h>

	using namespace cv;
	using namespace std;


	int main()
	{
		char keyPressed;
		VideoCapture _CameraDevice;
		Mat InputImage;

		_CameraDevice.open(1);

		_CameraDevice.set(CV_CAP_PROP_CONVERT_RGB, false);
		//_CameraDevice.set(CV_CAP_PROP_FOURCC,1);

		while (1)
		{
			_CameraDevice >> InputImage; //Read the input image

			if (InputImage.empty()) //Check for the vlid image
			{
				cout << "No frame grabbed!!, check whether the camera is free!!" << endl;
				break;
			}

			//cout << InputImage << endl;

			namedWindow("Camera Frame", WINDOW_AUTOSIZE);
			imshow("Camera Frame", InputImage);

			keyPressed = waitKey(1); //Waits for a user input to quit the application

			if (keyPressed == 27 || keyPressed == 'q' || keyPressed == 'Q')
			{
				destroyAllWindows();
				break;
			}
		}

		//Release the devices
		_CameraDevice.release();

		cout << endl << "Press any Key to exit the application" << endl << endl;
		_getch();

		return 0;
	}
screen_shot

**Comment by alalek** [comment]
@Guille129 This patch is about V2L, than means "Video for Linux". But your problem is Windows-related.
You should post your question on Users OpenCV Q/A forum: http://answers.opencv.org

**Pull request merged**